e2etests/test/empty_messages.js

```diff
@@ -0,0 +1,210 @@
+var DeviceIdentityHelper = require('./device_identity_helper.js');
+var Rendezvous = require('./rendezvous_helper.js').Rendezvous;
+
+var deviceAmqp = require('azure-iot-device-amqp');
+var deviceMqtt = require('azure-iot-device-mqtt');
+var deviceHttp = require('azure-iot-device-http');
+
+var hubConnectionString = process.env.IOTHUB_CONNECTION_STRING;
+
+[
+  deviceHttp.Http,
+  deviceAmqp.Amqp,
+  deviceAmqp.AmqpWs,
+  deviceMqtt.MqttWs,
+  deviceMqtt.Mqtt
+].forEach(function (deviceTransport) {
+  empty_message_tests(deviceTransport, DeviceIdentityHelper.createDeviceWithSymmetricKey);
+});
+
+function empty_message_tests(deviceTransport, createDeviceMethod) {
+  describe('Over ' + deviceTransport.name + ' using device/service clients c2d', function () {
+    this.timeout(60000);
+
+    var serviceClient, deviceClient;
+    var provisionedDevice;
+
+    before(function (beforeCallback) {
+      createDeviceMethod(function (err, testDeviceInfo) {
+        provisionedDevice = testDeviceInfo;
+        beforeCallback(err);
+      });
+    });
+
+    after(function (afterCallback) {
+      DeviceIdentityHelper.deleteDevice(provisionedDevice.deviceId, afterCallback);
+    });
+
+    beforeEach(function () {
+      serviceClient = serviceSdk.Client.fromConnectionString(hubConnectionString);
+      deviceClient = createDeviceClient(deviceTransport, provisionedDevice);
+    });
+
+    afterEach(function (done) {
+      closeDeviceServiceClients(deviceClient, serviceClient, done);
+    });
+
+    it('Service sends a C2D message with an empty payload and it is received by the device', function (done) {
+      var deviceClientParticipant = 'deviceClient';
+      var serviceClientParticipant = 'serviceClient';
+      var testRendezvous = new Rendezvous(done);
+      testRendezvous.imIn(deviceClientParticipant);
+      this.timeout(18000);
+      //
+      // The size of non payload is calculated based on the value of the messageId value size,
+      // the expiryTimeUtc value size and then on the application property value size plus one
+      // for the application property name.
+      //
+      var uuidData = uuid.v4();
+      var message = new Message('');
+      message.messageId = uuidData;
+      message.expiryTimeUtc = Date.now() + 60000; // Expire 60s from now, to reduce the chance of us hitting the 50-message limit on the IoT Hub
+
+      var foundTheMessage = false;
+      deviceClient.open(function (openErr) {
+        debug('device has opened.');
+        if (openErr) {
+          done(openErr);
+        } else {
+          debug('about to connect a listener.');
+          deviceClient.on('message', function (msg) {
+            debug('received a message');
+            //
+            // Make sure that the message we are looking at is one of the messages that we just sent.
+            //
+            if (msg.messageId === uuidData) {
+              foundTheMessage = true;
+            }
```

Comment:
nit: `foundTheMessage = (msg.messageId === uuidData)`

Reply:
Done.

---
In reply to: [227564328](https://github.com/Azure/azure-iot-sdk-node/pull/401#discussion_r227564328) [](ancestors = 227564328)